src/main/java/com/iota/iri/service/API.java

```diff
@@ -804,35 +718,26 @@ public static void incEllapsedTime_PoW(long ellapsedTime) {
         return elements;
     }
 
-    private AbstractResponse addNeighborsStatement(final List<String> uris) throws URISyntaxException {
-
+    private AbstractResponse addNeighborsStatement(final List<String> uris) {
         int numberOfAddedNeighbors = 0;
-        for (final String uriString : uris) {
-            final URI uri = new URI(uriString);
 
-            if ("udp".equals(uri.getScheme()) || "tcp".equals(uri.getScheme())) {
-                log.info("Adding neighbor: "+uriString);
-                // 3rd parameter true if tcp, 4th parameter true (configured tethering)
-                final Neighbor neighbor;
-                switch(uri.getScheme()) {
-                    case "tcp":
-                        neighbor = new TCPNeighbor(new InetSocketAddress(uri.getHost(), uri.getPort()),true);
-                        break;
-                    case "udp":
-                        neighbor = new UDPNeighbor(new InetSocketAddress(uri.getHost(), uri.getPort()), instance.node.getUdpSocket(), true);
-                        break;
-                    default:
-                        return ErrorResponse.create("Invalid uri scheme");
-                }
-                if (!instance.node.getNeighbors().contains(neighbor)) {
-                    instance.node.getNeighbors().add(neighbor);
-                    numberOfAddedNeighbors++;
-                }
+        for (final String uriString : uris) {
+            final URI uri = getValidUri(uriString);
+            log.info("Adding neighbor: "+uriString);
+            // 3rd parameter true if tcp, 4th parameter true (configured tethering)
+            final Neighbor neighbor;
+            if ("tcp".equals(uri.getScheme())) {
+                neighbor = new TCPNeighbor(new InetSocketAddress(uri.getHost(), uri.getPort()), true);
+            } else {
```

Comment:
probably not a good assumption to assume the else case is going to be "udp", better be specific.

Reply:
nvm, I see how `getValidUri` would throw an exception